digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/MunicipalityApi.scala

```diff
@@ -1,1132 +1,390 @@
-        }
-        case Obstacles.typeId => extractPropertyValue(getAssetNameProp(assetTypeId).head, prop, propertyValueToInt).asInstanceOf[Seq[Int]].foreach {
-          value => if (!Seq(1,2).contains(value))
-            halt(BadRequest(s"The property values for the property with name obstacleType are not valid."))
-        }
-        case PedestrianCrossings.typeId  => extractPropertyValue(getAssetNameProp(assetTypeId).head, prop, propertyValueToInt).asInstanceOf[Seq[Int]].foreach { value =>
-          if (!Seq(0,1).contains(value))
-            halt(BadRequest(s"The property values for the property with name hasPedestrianCrossing are not valid."))
-        }
-        case RailwayCrossings.typeId  =>
-            extractPropertyValue("safetyEquipment", prop, propertyValueToInt).asInstanceOf[Seq[Int]].foreach { value =>
-            if (!Seq(1,2,3,4,5).contains(value))
-              halt(BadRequest(s"The property values for the property with name safetyEquipment is not valid."))
-          }
-        case TrafficLights.typeId  => extractPropertyValue(getAssetNameProp(assetTypeId).head, prop, propertyValueToInt).asInstanceOf[Seq[Int]].foreach { value =>
-          if (!Seq(0,1).contains(value))
-            halt(BadRequest(s"The property values for the property with name hasTrafficLight are not valid."))
-        }
-        case assetType7restrictions if Maximum7Restrictions.contains(assetType7restrictions) =>
-          extractPropertyValue(getAssetNameProp(assetTypeId).head, prop, propertyValueToInt).asInstanceOf[Seq[Int]].foreach { value =>
-            if(!value.toString.forall(_.isDigit) || value <= 0)
-              halt(BadRequest(s"The property values for " + AssetTypeInfo.apply(assetTypeId).label + " asset type are not valid."))
-          }
-        case RoadWidth.typeId => extractPropertyValue(getAssetNameProp(assetTypeId).head, prop, propertyValueToInt).asInstanceOf[Seq[Int]].foreach { value =>
-          if(!value.toString.forall(_.isDigit) || value <= 0)
-            halt(BadRequest(s"The property values for "  + AssetTypeInfo.apply(assetTypeId).label + " asset type are not valid."))
+            featuresWithoutIds.sum
+          case None =>
+            awsDao.updateDatasetStatus(dataset.datasetId, DatasetStatus.ErrorsFeatures.value)
+            1
         }
-        case NumberOfLanes.typeId => extractPropertyValue(getAssetNameProp(assetTypeId).head, prop, propertyValueToInt).asInstanceOf[Seq[Int]].foreach { value =>
-          if(!value.toString.forall(_.isDigit) || value <= 0)
-            halt(BadRequest(s"The property values for "  + AssetTypeInfo.apply(assetTypeId).label + " asset type are not valid."))
+        missingIds
+      }).sum
+      featuresWithoutIds
+    } else {
+      0
+    }
+  }
+
+  def updateDataset(dataset: Dataset) = {
+    if (awsDao.getDatasetStatus(dataset.datasetId) != DatasetStatus.FeatureRoadlinksDontMatch.value) {
```

Comment:
If we have a method of how already validate this why we don´t save the return of the first validation in a variable and only execute this method if it is valid, with that we avoid this new get to database!